hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotDiffManager.java

```diff
@@ -0,0 +1,328 @@
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.utils.db.cache.CacheKey;
+import org.apache.hadoop.hdds.utils.db.cache.CacheValue;
+import org.apache.hadoop.ozone.OzoneConfigKeys;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.OmSnapshot;
+import org.apache.hadoop.ozone.om.OmTestManagers;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmVolumeArgs;
+import org.apache.hadoop.ozone.om.helpers.SnapshotDiffJob;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse;
+import org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse.JobStatus;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authentication.client.AuthenticationException;
+import org.apache.ozone.test.GenericTestUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.rocksdb.RocksDBException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationFactor.ONE;
+import static org.apache.hadoop.ozone.om.OmSnapshotManager.DELIMITER;
+
+/**
+ * Tests for {@link SnapshotDiffManager}.
+ */
+public class TestSnapshotDiffManager {
+
+  private static final String VOLUME = "vol";
+  private static final String BUCKET = "bucket";
+
+  private static File metaDir;
+  private static OzoneManager ozoneManager;
+  private static OMMetadataManager omMetadataManager;
+  private static SnapshotDiffManager snapshotDiffManager;
+  private static PersistentMap<String, SnapshotDiffJob> snapDiffJobTable;
+
+  @BeforeAll
+  public static void init() throws AuthenticationException,
+      IOException, RocksDBException {
+    metaDir = GenericTestUtils.getRandomizedTestDir();
+    if (!metaDir.exists()) {
+      Assertions.assertTrue(metaDir.mkdirs());
+    }
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.set(OzoneConfigKeys.OZONE_METADATA_DIRS,
+        metaDir.getAbsolutePath());
+
+    OmTestManagers omTestManagers = new OmTestManagers(conf);
+    ozoneManager = omTestManagers.getOzoneManager();
+    omMetadataManager = omTestManagers.getMetadataManager();
+
+    snapshotDiffManager = ozoneManager
+        .getOmSnapshotManager().getSnapshotDiffManager();
+    snapDiffJobTable = snapshotDiffManager.getSnapDiffJobTable();
+
+    createVolumeAndBucket();
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    FileUtil.fullyDelete(metaDir);
+  }
+
+  @Test
+  public void testListSnapshotDiffJobs()
```

Comment:
nit: It would be nicer if you convert this test to [parameterized test](https://www.baeldung.com/parameterized-tests-junit-5). Repetitive code can be avoid with that.

Sample test: https://github.com/apache/ozone/blob/9034434285f3877246b3ebc6a7513c082931b279/hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java#L263

Reply:
done.